### DISCUSSION TITLE
SPUUpdaterDelegate feedURLString not working

### DISCUSSION THREAD

**Discussion by Alec-4Privacy** [sparkle-project/Sparkle#2366] [Q&A] [answered]
### Sparkle's output from Console.app
Error: You must specify the URL of the appcast as the SUFeedURL key in either the Info.plist, or with -[SPUUpdaterDelegate feedURLStringForUpdater:] for app! (null) (URL (null))
```

```

### Steps to reproduce the behavior

The application is a flutter application but I am not sure that matters. In the MacOS swift project, we have added the Sparkle framework via pods. 

AppDelegate looks like this: 
`
class AppDelegate: FlutterAppDelegate {
    let updaterController: SPUStandardUpdaterController
    
    override func applicationShouldTerminateAfterLastWindowClosed(_ sender: NSApplication) -> Bool {
        return true
    }
    
    override init() {
        updaterController = SPUStandardUpdaterController(startingUpdater: true, updaterDelegate: CustomSPUUpdaterDelegate(), userDriverDelegate: nil)
    }
    
    override func applicationDidFinishLaunching(_ aNotification: Notification) {
        
        //Register for platform-specific method handler
        let controller : FlutterViewController = mainFlutterWindow?.contentViewController as! FlutterViewController
        
        let sparkleChannel = FlutterMethodChannel.init(name: "com.app/sparkle", binaryMessenger: controller.engine.binaryMessenger)
        sparkleChannel.setMethodCallHandler({
            (call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
            if call.method == "launchUpdaterUI" {
                self.updaterController.checkForUpdates(nil);
                result(nil);
            } else {
                result(FlutterMethodNotImplemented)
            }
        })
        
        super.applicationDidFinishLaunching(aNotification)
    }
}

`

CustomSPUUpdaterDelegate is as simple as it can get, it just returns a hardcoded string. We want to do more here but just testing the simplest scenario. 

`class CustomSPUUpdaterDelegate: NSObject, SPUUpdaterDelegate {
    func feedURLString(for updater: SPUUpdater) -> String? {
        return "https://app.blob.core.windows.net/artifacts/dev/appcast.xml";
    }
}`

**Comment by zorgiepoo** [comment, marked as answer]
One issue I see is your `AppDelegate` needs to keep the `CustomSPUUpdaterDelegate` alive as a property or instance variable. The updater weakly references it and doesn't strongly retain it.
